### PR TITLE
turned off all new rules that cause errors on rci

### PR DESCRIPTION
**Pull request by troyblank**
OK, since we updated a bunch of stuff I had to turn OFF a bunch of new rules that cause tons of errors on components. I also added notes around good ones that we should might want to turn on.

### CODE REVIEW

File: configs/react.js [hunk 1]
```diff
@@ -3,25 +3,34 @@ module.exports = {
         'eslint-config-airbnb/rules/react'
     ],
     rules: {
-        'react/jsx-uses-react': 'error',
-        'react/jsx-uses-vars': 'error',
-        'react/jsx-no-duplicate-props': 'error',
+        'jsx-quotes': 'error',
+        'no-extra-parens': 'off',
+        'react/button-has-type': 0,
+        'react/default-props-match-prop-types': 0, // up for consideration
```
Comment (jasonhale) [inline configs/react.js]: 👍

**Comment by troyblank** [comment]
@jasonhale  is this good to go in your eyes?

File: configs/react.js [hunk 2]
```diff
@@ -3,25 +3,34 @@ module.exports = {
...
+        'react/destructuring-assignment': 0, // up for consideration
+        'react/display-name': 'error',
+        'react/jsx-boolean-value': ['error', 'always'],
+        'react/jsx-curly-brace-presence': 0,
+        'react/jsx-filename-extension': [1, { extensions: ['.js', '.jsx'] }],
+        'react/jsx-key': 'error',
+        'react/jsx-no-duplicate-props': 0,
         'react/jsx-no-undef': 'error',
         'react/jsx-no-bind': 'error',
+        'react/jsx-one-expression-per-line': 0,
```
Comment (jasonhale) [inline configs/react.js]: Seems a little restrictive.  Consider `{ "allow": "literal" }`?

Comment (troyblank) [reply configs/react.js]: Sorry not sure how it's restrictive, it's turned off completely as it causes lint errors on rci-components. I would assume in the future we could turn it on with that inconsideration though.

Comment (jasonhale) [inline configs/react.js]: If allowed, this one i see as a little restrictive.  Maybe with an `{ "allow": "literal" }`

**Pull request merged**